spyder/widgets/editor.py

```diff
@@ -1309,12 +1311,22 @@ def select_savename(self, original_filename):
 
     def save_as(self, index=None):
         """Save file as..."""
+
         if index is None:
             # Save the currently edited file
+            if not self.get_stack_count():
+                return
             index = self.get_stack_index()
+
         finfo = self.data[index]
-        filename = self.select_savename(finfo.filename)
+
+        self.redirect_stdio.emit(False)
+        filename, _selfilter = getsavefilename(self, _("Save file"),
+                                               finfo.filename, get_edit_filters())
+        self.redirect_stdio.emit(True)

```

Comment:
We need a smarter way so that if I click on save us, by default I should get the selection to be the actual file extension I have.

You can make a test of this by creating a .txt or .yaml file and then click on save as....  we should not get 

somefile.txt.py or somefile.yaml... we should get somefile.txt and the appropriate selection on the combobox